src/core/thread/thread_netif.cpp

```diff
@@ -222,4 +224,96 @@ bool ThreadNetif::IsTmfMessage(const Ip6::MessageInfo &aMessageInfo)
     return rval;
 }
 
+void ThreadNetif::SetAddressCallback(otIp6AddressCallback aCallback, void *aCallbackContext)
```

Comment:
Why not adding this in `Netif` class instead?

Reply:
An `otInstance` can only have one `ThreadNetif`, however it may have multiple `Netif`. I think we will only notify address changes on `ThreadNetif`.

Reply:
Yes. But having the mechanism to register an address callback handler and get address added/removed notifications would be helpful for other `Netif` instances (if we ever need/use them). If we add this method to `Netif`, then `ThreadNetif` would also inherit it and we can use it from `otIp6SetAddressCallback()`. 

Currently all address management related APIs and logic is in `Netif` so adding the callback to `Netif` fits the current model.